src/Lumi/Components/Form/Validation.purs

```diff
@@ -58,15 +60,27 @@ type WarningValidator result =
 
 -- | A `Validator` which verifies that an input string is non-empty.
 nonEmpty :: String -> Validator String NonEmptyString
-nonEmpty name = note (name <> " is required") <<< NES.fromString
+nonEmpty name = nonEmpty' (name <> " is required")
+
+-- | `nonEmpty`, but the argument is the entire validation message.
+nonEmpty' :: String -> Validator String NonEmptyString
+nonEmpty' msg = note msg <<< NES.fromString
```

Comment:
Nice!!

Reply:
These have been bugging me for a while haha